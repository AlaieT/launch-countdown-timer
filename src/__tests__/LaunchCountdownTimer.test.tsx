import React from "react";
import { render, cleanup } from "@testing-library/react";

import LaunchCountdownTimer from "../components/LaunchCountdownTimer";

describe("LaunchCountdownTimer", () => {
  const props = {
    title: "WE'RE LAUNCHING SOON",
    deadline: "April 30, 2023",
    links: {
      facebook: "",
      pinterest: "",
      instagram: ""
    },
    background: {
      starsImage: "/",
      hillsImage: "/"
    }
  };

  describe("correct render", () => {
    it("should match snapshot", () => {
      expect(
        render(
          <LaunchCountdownTimer
            background={props.background}
            deadline={props.deadline}
            title={props.title}
            links={props.links}
          />
        ).asFragment()
      ).toMatchSnapshot();
    });
  });
});
