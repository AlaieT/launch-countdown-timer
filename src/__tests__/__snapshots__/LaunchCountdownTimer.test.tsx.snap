// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchCountdownTimer correct render should match snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1>
      WE'RE LAUNCHING SOON
    </h1>
  </header>
  <main>
    <div
      id="background"
    >
      <img
        alt="bg-stars"
        src="/"
      />
      <img
        alt="bg-hills"
        src="/"
      />
    </div>
    <div
      class="digit"
    >
      <div
        class="box"
      >
        <div
          class="top"
        >
          <p>
            03
          </p>
        </div>
        <div
          class="middle"
        >
          <div
            class="front"
          >
            <p>
              03
            </p>
          </div>
          <div
            class="back"
          >
            <p>
              03
            </p>
          </div>
        </div>
        <div
          class="bottom"
        >
          <p>
            03
          </p>
        </div>
      </div>
      <div
        class="label"
      >
        <p>
          days
        </p>
      </div>
    </div>
    <div
      class="digit"
    >
      <div
        class="box"
      >
        <div
          class="top"
        >
          <p>
            11
          </p>
        </div>
        <div
          class="middle"
        >
          <div
            class="front"
          >
            <p>
              11
            </p>
          </div>
          <div
            class="back"
          >
            <p>
              11
            </p>
          </div>
        </div>
        <div
          class="bottom"
        >
          <p>
            11
          </p>
        </div>
      </div>
      <div
        class="label"
      >
        <p>
          hours
        </p>
      </div>
    </div>
    <div
      class="digit"
    >
      <div
        class="box"
      >
        <div
          class="top"
        >
          <p>
            14
          </p>
        </div>
        <div
          class="middle"
        >
          <div
            class="front"
          >
            <p>
              14
            </p>
          </div>
          <div
            class="back"
          >
            <p>
              14
            </p>
          </div>
        </div>
        <div
          class="bottom"
        >
          <p>
            14
          </p>
        </div>
      </div>
      <div
        class="label"
      >
        <p>
          minuts
        </p>
      </div>
    </div>
    <div
      class="digit"
    >
      <div
        class="box"
      >
        <div
          class="top"
        >
          <p>
            30
          </p>
        </div>
        <div
          class="middle"
        >
          <div
            class="front"
          >
            <p>
              30
            </p>
          </div>
          <div
            class="back"
          >
            <p>
              30
            </p>
          </div>
        </div>
        <div
          class="bottom"
        >
          <p>
            30
          </p>
        </div>
      </div>
      <div
        class="label"
      >
        <p>
          seconds
        </p>
      </div>
    </div>
  </main>
  <footer>
    <ul>
      <li>
        <a
          aria-label="facebook link"
          href=""
          rel="noreferrer"
          target="_blank"
        >
          <div />
        </a>
      </li>
      <li>
        <a
          aria-label="pinterest link"
          href=""
          rel="noreferrer"
          target="_blank"
        >
          <div />
        </a>
      </li>
      <li>
        <a
          aria-label="instagram link"
          href=""
          rel="noreferrer"
          target="_blank"
        >
          <div />
        </a>
      </li>
    </ul>
  </footer>
</DocumentFragment>
`;
